server/bin/wsserver.py

```diff
@@ -47,7 +47,7 @@ def master(options, trackDirectLogger):
                 childFDs={0: 0, 1: 1, 2: 2, port.fileno(): port.fileno()},
                 env=os.environ)
 
-        # reactor.run()
```

Comment:
I guess you want the main thread to continue to execute to receive logging messages? But inital thread is only used for starting the workers, when that is done it has nothing more to do. I suggest that we do not do this change, we can even remove this row (it is a remnant from before it was multi-threaded.).

Reply:
the issue is with docker: docker needs some kind of blocking process, otherwise docker is thinking the process closed successfully and will terminate the container.
maybe spawning some threads internally would be better than spawning some completely new processes. what do you think?

Reply:
Make a docker/systemd switch to be blocking/non-blocking.
I for one would NOT want to run it in docker (sysadmin reasons) while others might (dev reasons) so adding a switch for the docker/systemd users would be preferred for me at least

Reply:
@hevisko you can always add an _&_ to execute the script to run it in the background.
I think the best is to make everything blocking inside and if you want to run the process in the background you just need an _&_ to move it there.
Adding code which is easier handled on system level is not a nice way to fix discussion like this.

Reply:
The suggested change can be done, but in that case we need to remove the row "port.stopReading()" to enable the master process to actually do something (otherwise that process will not listen to incoming request). 

We should probably also do -1 on the for loop range to make sure we have one process per cpu.

Reply:
> @hevisko you can always add an _&_ to execute the script to run it in the background. I think the best is to make everything blocking inside and if you want to run the process in the background you just need an _&_ to move it there. Adding code which is easier handled on system level is not a nice way to fix discussion like this.

That would be how devs see the world, while sysadmins know that a '&' also means that you need to now (outside) needs to tell systemd not to kill the group, and to do other things like first disabling the TTY, else you'll have that dongling around that has other funs - I'm thinking about hands off starting.